apps/dashboard/lib/trpc/routers/identity/search.ts

```diff
@@ -0,0 +1,81 @@
+import { db } from "@/lib/db";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { ratelimit, requireWorkspace, t, withRatelimit } from "../../trpc";
+import { IdentityResponseSchema } from "./query";
+
+const LIMIT = 5;
+
+const SearchIdentitiesResponse = z.object({
+  identities: z.array(IdentityResponseSchema),
+});
+
+export const searchIdentities = t.procedure
+  .use(requireWorkspace)
+  .use(withRatelimit(ratelimit.read))
+  .input(
+    z.object({
+      query: z
+        .string()
+        .min(1, "Search query is required")
+        .max(255, "Search query is too long")
+        .trim()
+        .refine((query) => !/^\s+$/.test(query), "Search query cannot be only whitespace"),
```

Comment:
If you reorder this to put trim() first, and then check the min length, you can get rid of the refine() since trim will remove any trailing or leading whitespace, and if it's only whitespace, it won't pass the min length validation after trim executes.